mk/subdir.mk

```diff
@@ -63,11 +63,12 @@ define process-subdir-gensrcs-helper
 # $2 full path and name of generated source file
 # $3 full path and name of object file compiled from source file
 # $4 full path to out directory
+# $5 true if $2 must be generated before $(sm) starts building (e.g., .h file)
 
 gen-srcs			+= $2
 oname				:= $3
 
-FORCE-GENSRC$(sm): $2
+FORCE-GENSRC$(sm): $(if $(filter true,$5),$2,)
```

Comment:
In these makefiles we're normally using `y` instead of `true`.

Reply:
Right, I'll fix that.